ui/src/message_popup/author/questions_page.jsx

```diff
@@ -59,18 +66,37 @@ export default React.createClass({
     return questionString.split(text).length;
   },
   
+  onQuestionsReceived(err, response){
+    const allDatabaseQuestions = JSON.parse(response.text).row;
+    if(allDatabaseQuestions === undefined) {
+      this.setState({
+        allDatabaseQuestions: {
+          currentQuestions: allQuestions, 
+          archivedQuestions: allArchivedQuestions
+        }, 
+        currentQuestions: allQuestions, 
+        archivedQuestions: allArchivedQuestions});
+      return;
+    }
+    this.setState({
+      allDatabaseQuestions: allDatabaseQuestions, 
+      currentQuestions: allDatabaseQuestions.currentQuestions, 
+      archivedQuestions: allDatabaseQuestions.archivedQuestions
+    });
+  },
+
   onNewQuestion(){
     Routes.navigate(Routes.messagePopupAuthorQuestionsNewPath());
   },
 
   onSearchBarChange(event){
     const value = event.target.value.toLowerCase().trim();
     if(value === ''){
-      this.setState({questions: allQuestions});
+      this.setState({currentQuestions: this.state.allDatabaseQuestions.currentQuestions});
       return;
     }
-    const questions = allQuestions.filter(questionOriginal => this.searchQuestionForText(value, questionOriginal)).sort(questionOriginal => this.countTextOccurrences(value, questionOriginal));
-    this.setState({questions});
+    const questions = this.state.allDatabaseQuestions.currentQuestions.filter(questionOriginal => this.searchQuestionForText(value, questionOriginal)).sort(questionOriginal => this.countTextOccurrences(value, questionOriginal));
+    this.setState({currentQuestions: questions});
```

Comment:
Previously, there were two different concepts: `allQuestions` and `questions`.  Now that we're updating this and the set of questions is fetched from the server and can change we'll want to keep these as distinct things.

I should have noticed this previously, but in React components in general, filtering and other computations should be in the `render` path whenever possible.  In this case we can compute the list to show from `currentQuestions` and the search text.  If this method stores the search text in component state, then we can describe this minimally with only two concepts, and the filtered list can be computed from them.